src/cargo/util/config.rs

```diff
@@ -100,7 +101,7 @@ impl Config {
         }
     }
 
-    pub fn default() -> CargoResult<Config> {
+    pub fn default(cargo_exe: PathBuf) -> CargoResult<Config> {
```

Comment:
Could we, instead of passing an argument to `default` method, call `std::env::args` here? `default` method with an  argument looks funny :) 

Also, perhaps we should try to use `current_exec` first, and only then fall-back to using `argv[0]`? I don't have any solid arguments pro or contra, but using `current_exec` seems appropriate in most cases.

Reply:
Ah yeah I agree with @matklad that using `current_exe` first and then falling back to `env::args` is probably the best strategy here